csharp/src/Ice/ConnectionFactory.cs

```diff
@@ -545,14 +543,15 @@ public void Remove(Connection connection)
 
         public override string ToString() => _acceptor.ToString()!;
 
-        internal AcceptorIncomingConnectionFactory(ObjectAdapter adapter, Endpoint endpoint, string serverName)
+        internal AcceptorIncomingConnectionFactory(ObjectAdapter adapter, Endpoint endpoint)
         {
             _communicator = adapter.Communicator;
             _adapter = adapter;
             _acceptor = endpoint.Acceptor(this, _adapter);
-
             Endpoint = _acceptor.Endpoint;
-            PublishedEndpoint = Endpoint.GetPublishedEndpoint(serverName);
+
+            // If Endpoint is an IP endpoint, it must have an IP address, not a DNS name.
+            Debug.Assert((Endpoint as IPEndpoint)?.Address != IPAddress.None);
```

Comment:
Shouldn't transport specific checks go the transport implementation instead?

Reply:
Yes, you're right, and this leads to further simplification as TcpAcceptor and UdpSocket can only be constructed with IPEndpoint that are not DNS names.

Fixed.